packages/react/src/Trans.tsx

```diff
@@ -69,19 +73,43 @@ export function Trans(props: TransProps) {
   } else if (typeof render === "string") {
     // Built-in element: h1, p
     return React.createElement(render, {}, translation)
-  } else if (React.isValidElement(render)) {
+  } else if (isClassComponent(render as TransRenderPropsType) || isFunctionalComponent(render)) {
+    // We apply the logic of React of:
+    // - Don't call function components. Render them.
+    // If we just use render(), as functional component,
+    // will "work" but not the way you'd expect and it could behave in unexpected ways as you make changes
+    return React.createElement(render as React.ElementType<TransRenderProps>, {
+      id,
+      translation,
+      message
+    })
+  } else if (React.isValidElement(render)) {
     // Element: <p className="lear' />
     return React.cloneElement(render, {}, translation)
   }
 
   // Component: (props) => <a title={props.translation}>x</a>
-  return React.createElement(render as React.ElementType<TransRenderProps>, {
+  return render({
```

Comment:
any function is `isFunctionalComponent`, so you will never execute this line

Reply:
I tough only functions that returns "internally" React.createElement are functional components, we cant render a functional component as a simple function, Wont work correctly